### DISCUSSION TITLE
performance comparison theia extension vs theia plugin cs vsCode plugin

### DISCUSSION THREAD

**Discussion by pilzandreas** [eclipse-theia/theia#9633] [General]
I wonder if anyone has experience with all three ways of extending theia - extension/plugin/vsCode-plugin and the related performance (especially with Web/BrowserView based plugins).

To me performance of all three options seems pretty equal - or at least no big disadvantage staying with a vsCode-Plugin comapred to a Theia extension...

**Comment by vince-fugnitto** [comment]
@pilzandreas I have not noticed any performance difference between the different ways (theia extension, theia plugin, and vscode extensions), but it may factor in on how you deploy your application, and if you make use of remote plugins or not (something `che` [does](https://github.com/eclipse-theia/theia/issues/6215#issuecomment-533013757)).

The main arguments are with what you require:
- `theia extensions` are build-time components that offer you more control, and allow you to extend, or override default functionality provided by the framework
- `vscode plugins`: allow you to use the simple vscode api, and you can target multiple applications (theia, vscode, vscodium, coder...)
- `theia plugins`: similar to vscode plugins but for theia-based applications (I haven't really used these).

**Reply by pilzandreas** [reply]
Thank You!
